tools/pythonpkg/tests/fast/spark/test_spark_join.py

```diff
@@ -328,3 +328,57 @@ def test_self_join(self, dataframe_a):
             Row(emp_id=5, name='Brown', superior_emp_id=2, superior_emp_name='Rose'),
             Row(emp_id=6, name='Brown', superior_emp_id=2, superior_emp_name='Rose'),
         ]
+
+    def test_join_with_using_clause(self, spark, dataframe_a):
+        dataframe_a = dataframe_a.select('name', 'year_joined')
+
+        df = dataframe_a.alias('df1')
+        df2 = dataframe_a.alias('df2')
+        res = df.join(df2, ['name', 'year_joined']).sort('name', 'year_joined')
+        res = res.collect()
+        assert res == [
+            Row(name='Brown', year_joined='2010'),
+            Row(name='Brown', year_joined='2010'),
+            Row(name='Brown', year_joined='2010'),
+            Row(name='Brown', year_joined='2010'),
+            Row(name='Jones', year_joined='2005'),
+            Row(name='Rose', year_joined='2010'),
+            Row(name='Smith', year_joined='2018'),
+            Row(name='Williams', year_joined='2010'),
+        ]
+
+    def test_join_with_common_column(self, spark, dataframe_a):
+        dataframe_a = dataframe_a.select('name', 'year_joined')
+
+        df = dataframe_a.alias('df1')
+        df2 = dataframe_a.alias('df2')
+        res = df.join(df2, df.name == df2.name).sort('df1.name')
+        res = res.collect()
+        assert (
+            str(res)
+            == "[Row(name='Brown', year_joined='2010', name='Brown', year_joined='2010'), Row(name='Brown', year_joined='2010', name='Brown', year_joined='2010'), Row(name='Brown', year_joined='2010', name='Brown', year_joined='2010'), Row(name='Brown', year_joined='2010', name='Brown', year_joined='2010'), Row(name='Jones', year_joined='2005', name='Jones', year_joined='2005'), Row(name='Rose', year_joined='2010', name='Rose', year_joined='2010'), Row(name='Smith', year_joined='2018', name='Smith', year_joined='2018'), Row(name='Williams', year_joined='2010', name='Williams', year_joined='2010')]"
+        )
+
+    @pytest.mark.xfail(
+        reason="The base relation in 'first_join' and 'second_join' have the same alias, causes binder exception"
+    )
+    def test_join_on_joined_data_error(self, spark):
+        df = spark.createDataFrame([(2, "Alice"), (5, "Bob")]).toDF("age", "name")
+        df2 = spark.createDataFrame([Row(height=80, name="Tom"), Row(height=85, name="Bob")], ["height", "name"])
+
+        first_join = df.join(df2, "name")
+        second_join = df.join(df2, "name")
+        third_join = first_join.join(second_join, "name")
+
+        third_join.show()
```

Comment:
This is point number 5 of the list of issues
@Mytherin it complains about duplicate alias, not sure how this should be fixed
```
duckdb.duckdb.BinderException: Binder Error: Duplicate alias "unnamed_relation_4c60c14f3e43605a" in query!
```

Reply:
All aliases are available after joins normally which is why this happens, e.g.:

```sql
CREATE TABLE df(name VARCHAR);
CREATE TABLE df2(name VARCHAR);
SELECT df.name, df2.name FROM df JOIN df2 USING (name);
┌─────────┬─────────┐
│  name   │  name   │
│ varchar │ varchar │
├─────────┴─────────┤
│      0 rows       │
└───────────────────┘
```

Joins preserve all child aliases, unlike subqueries that flatten all child aliases into a single alias.

Now for the Spark API we might want to wrap joins in subqueries, which would allow us to circumvent this. The only problem is this results in odd results when dealing with duplicate names across the relations, e.g.:

```sql
D CREATE TABLE df(id INT, name VARCHAR);
D CREATE TABLE df2(id INT, name VARCHAR);
D SELECT * FROM (FROM df JOIN df2 USING (id));
┌───────┬─────────┬─────────┐
│  id   │  name   │ name_1  │
│ int32 │ varchar │ varchar │
├───────┴─────────┴─────────┤
│          0 rows           │
└───────────────────────────┘
```

As you can see the original information (whether or not `name` came from `df` or from `df2`) is lost.

I'm not sure what the best solution is here - we could perhaps also automatically modify the alias in case of a conflict as a setting, and use that in the Spark API? I'm not sure what Spark does here and how it deals with conflicts in join aliases and availability of columns afterwards.

Reply:
Let's start off by just pushing a subquery after a join to flatten the namespace similar to what Spark does. We will have different behavior than Spark when it comes to duplicate names (as we will disambiguate them) but we can leave that as-is for now